### DISCUSSION TITLE
Send command to multiple hub at the same time.

### DISCUSSION THREAD

**Discussion by PastaDental** [mvp/uhubctl#355] [Q&A] [answered]
Hello all,

Is it possible to send a command to multiple hub at the same time the same way as I can to multiple ports on a hub.  I'm running multiple hubs (4) on a RP3 B+.

Right now I run the command for each hub one after the other and this creates a pretty long delay when sending the command.  I know I can can to multiple ports on the same hub.

os.sytem ("sudo ./uhubctl -a 1 -l 1-1.3 -p 1,2")
os.sytem ("sudo ./uhubctl -a 1 -l 1-1.4.4.4 -p 1")

How can I run these two lines at the same time?

**Comment by mvp** [comment, marked as answer]
That would require completely changing command line syntax. And even if it did support it, it wouldn't make it any faster as uhubctl would process it serially.

Instead, I recommend you to call uhubctl from multiple threads of your application targeting separate hubs - that would make it get your result faster overall.

**Reply by PastaDental** [reply]
Thanks for the reply mvp.  I will try this for sure, I didn't think of this.

**Comment by PastaDental** [comment]
For future reference.  This works perfectly to open multiple ports on multiple hub at the same time.

import os
import _thread
from gpiozero import Button
from signal import pause

def BLUEon():
    _thread.start_new_thread( BLUEon1, ("Un", 1, ) )
    _thread.start_new_thread( BLUEon2, ("Deux", 1 ) )
    _thread.start_new_thread( BLUEon3, ("Trois", 2 ) )
    _thread.start_new_thread( BLUEon4, ("Quatre", 2 ) )
    _thread.start_new_thread( BLUEon5, ("Cinq", 3, ) )

def BLUEoff():
    _thread.start_new_thread( BLUEoff1, ("Un", 1, ) )
    _thread.start_new_thread( BLUEoff2, ("Deux", 1 ) )
    _thread.start_new_thread( BLUEoff3, ("Trois", 2, ) )
    _thread.start_new_thread( BLUEoff4, ("Quatre", 2 ) )
    _thread.start_new_thread( BLUEoff5, ("Cinq", 3, ) )
    
def BLUEon1( threadName, delay ):
    os.system ("uhubctl -a 1 -l 1-1.5.4 -p 1,3")
def BLUEon2( threadName, delay ):
    os.system ("uhubctl -a 1 -l 1-1.5 -p 2")
def BLUEon3( threadName, delay ):
    os.system ("uhubctl -a 1 -l 1-1.3 -p 1") 
def BLUEon4( threadName, delay ): 
    os.system ("uhubctl -a 1 -l 1-1.3.4.4 -p 1") 
def BLUEon5( threadName, delay ):
     os.system ("uhubctl -a 1 -l 1-1.4 -p 2")
def BLUEoff1( threadName, delay ):
    os.system ("uhubctl -a 0 -l 1-1.5.4 -p 1,3")
def BLUEoff2( threadName, delay ):   
    os.system ("uhubctl -a 0 -l 1-1.5 -p 2") 
def BLUEoff3( threadName, delay ):
        os.system ("uhubctl -a 0 -l 1-1.3 -p 1") 
def BLUEoff4( threadName, delay ):
    os.system ("uhubctl -a 0 -l 1-1.3.4.4 -p 1")
def BLUEoff5( threadName, delay ):
    os.system ("uhubctl -a 0 -l 1-1.4 -p 2") 
 
SwitchBleu = Button(6)
SwitchBleu.when_pressed = BLUEon
SwitchBleu.when_released = BLUEoff
pause()
